Tasks/Gradle/package.json

```diff
@@ -8,8 +8,12 @@
     "url": "https://github.com/Microsoft/vso-agent-tasks/issues"
   },
   "dependencies": {
+    "codeanalysis-common": "file:../../_build/Tasks/Common/codeanalysis-common",
```

Comment:
Why are these entries here?  I'm not sure I've seen this pattern in other tasks.

Reply:
We do need these, or at least we will going forward. I have a change out to migrate to npm5 and the way the build process works will require all common dependencies to be included. We also need them in make.json so that those projects get built.

In the current codebase I think they weren't 100% required as we would use make.json to build the common package, cd to the Task dir, then npm install each common package. In the new setup we use make.json to build the common packages but only do one npm install in the Task dir which uses the dependencies section to install normal npm packages as well as the common ones.

Not having these in the past was somewhat hacky and went against how packages should be managed. We leaned on our own scripts too much, IMO :) Though there could have been a historical reason for that.

Reply:
That's right, we don't need them in the current codebase. However it seems a good improvement, so I accepted the change from the SonarSource team.

Reply:
👍